test/e2e/metering_manual_install_tests.go

```diff
@@ -107,6 +108,149 @@ func testManualMeteringInstall(
+			}
+			ctx.Logger.Debugf("Created the %s nfs_server %s nfs_service %s nfs_storageclass", nfsManifestItems["nfsServer"], nfsManifestItems["nfsService"].resourceFile, nfsManifestItems["nfsStorageclass"].resourceFile)
+			if action == "GET" {
+				_, err := ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+				if err != nil && !apierrors.IsNotFound(err) {
+					return nil
+				}
+			}
+		}
+	}
+
+	//  - Wait for service to populate spec.ClusterIP (poll.Wait functionality)
+	// poll has a closure that polls for the server IP to get populated
+	var service, err = ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+
+	// if theres an error don't poll
+	if err != nil {
+		err := wait.Poll(3*time.Second, 5*time.Minute, func() (done bool, err error) {
+			service, err = ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+			return false, err
+		}
+		// clusterIP is populated, stop polling
+		return service.Spec.ClusterIP != "", nil
+	})
+}
+
+
+
+
+
+	// 	if filepathNFSPersistentvolume.spec.nfs.server != ""
+	// 	{
+	// 	clusterIP := filepathNFSPersistentvolume.spec.nfs.server
+	// 	return clusterIP
+	// }
+	// 	return pollInternal(poller(interval, timeout), condition)
+	// }
+
+	//  - Decode the NFS PV YAML manifest
+	//get a go object
+	// after we get clusterIP, save it in a var, pvc.spec.nfs.server = val
+	filepathNFSPersistentvolume := "test/testhelpers/nfs_persistentvolume.yaml"
+	err := deploy.DecodeYAMLManifestToObject(filepathNFSPersistentvolume, &nfs_persistentvolume)
+	if err != nil {
+		return "", fmt.Errorf("failed to decode the %s YAML service manifest: %v", filepathNFSPersistentvolume, err)
+
+		// Update the NFS PV configuration w/ the populated spec.ClusterIP from the NFS Service
+		clusterIP := service.spec.nfs.server
+	}
+
+	_, err = ctx.Client.CoreV1().Namespaces().Create(context.Background(), &nfs_persistentvolume, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	ctx.Logger.Debugf("Created the %s nfs_clusterIP", clusterIP)
+
+	return err
+}
+
+func createNamespace(ctx *deployframework.DeployerCtx) (*corev1.Namespace, error) {
+	namespace, err := ctx.Client.CoreV1().Namespaces().Get(context.Background(), ctx.Namespace, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return namespace, err
+	}
+	if apierrors.IsNotFound(err) {
+		namespace := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: ctx.Namespace,
+				Labels: map[string]string{
+					"name": ctx.Namespace + "-" + testingNamespaceLabel,
+				},
+			},
+		}
+		_, err = ctx.Client.CoreV1().Namespaces().Create(context.Background(), namespace, metav1.CreateOptions{})
+		if err != nil {
+			return namespace, nil
+		}
+		ctx.Logger.Debugf("Created the %s namespace", ctx.Namespace)
+	}
+	return namespace, nil
+}
```

Comment:
This is fine to keep as a helper function as we can use it in other spots too, but we wouldn't want to pass an entire `deployframework.DeployerCtx` structure as a parameter as we're only interested in the actual name of the ctx.Namespace string. We would want to also modify the name of this, replacing `createNamespace` with something like `createTestingNamespace` and go from there.